// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.cmrt.sdk.job_service.v1;

option java_multiple_files = true;
option java_outer_classname = "JobServiceConfigurationKeysProto";
option java_package = "com.google.cmrt.sdk.job_service.v1";

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

// Configuratiion keys for client level.
enum ClientConfigurationKeys {
  // Invalid key.
  UNKNOWN_JOB_CLIENT_CONFIGURATION_KEY = 0;
  // Optional. If not set, use the default value 2. The number of
  // CompletionQueues for the server.
  // https://grpc.github.io/grpc/cpp/classgrpc_1_1_server_builder.html#aff66bd93cba7d4240a64550fe1fca88d
  CMRT_JOB_CLIENT_COMPLETION_QUEUE_COUNT = 1;
  // Optional. If not set, use the default value 2. Minimum number of polling
  // threads for the server.
  // https://grpc.github.io/grpc/cpp/classgrpc_1_1_server_builder.html#aff66bd93cba7d4240a64550fe1fca88d
  CMRT_JOB_CLIENT_MIN_POLLERS = 2;
  // Optional. If not set, use the default value 5. Maximum number of polling
  // threads for the server.
  // https://grpc.github.io/grpc/cpp/classgrpc_1_1_server_builder.html#aff66bd93cba7d4240a64550fe1fca88d
  CMRT_JOB_CLIENT_MAX_POLLERS = 3;
  // Optional. If not set, use the default value 2.
  CMRT_JOB_CLIENT_CPU_THREAD_COUNT = 4;
  // Optional. If not set, use the default value 100000.
  CMRT_JOB_CLIENT_CPU_THREAD_POOL_QUEUE_CAP = 5;
  // Optional. If not set, use the default value 2.
  CMRT_JOB_CLIENT_IO_THREAD_COUNT = 6;
  // Optional. If not set, use the default value 100000.
  CMRT_JOB_CLIENT_IO_THREAD_POOL_QUEUE_CAP = 7;
  // Required. The name of the job queue to use.
  CMRT_JOB_CLIENT_JOB_QUEUE_NAME = 8;
  // Required. The name of the job table to use.
  CMRT_JOB_CLIENT_JOB_TABLE_NAME = 9;
  // Required for GCP. The name of the Spanner Instance to use.
  CMRT_GCP_JOB_CLIENT_SPANNER_INSTANCE_NAME = 10;
  // Required for GCP. The name of the Spanner Database to use.
  CMRT_GCP_JOB_CLIENT_SPANNER_DATABASE_NAME = 11;
  // Optional. Retry interval for reading job from database.
  // If not set, use the default value 501.
  CMRT_JOB_CLIENT_READ_JOB_RETRY_INTERVAL_IN_MS = 12;
  // Optional. Maximum retry count for reading job from database.
  // If not set, use the default value 8.
  CMRT_JOB_CLIENT_READ_JOB_MAX_RETRY_COUNT = 13;
}
